src/apply.rs

```diff
@@ -18,35 +18,68 @@ pub fn script(
     app_instance: &AppInstance,
     manifests_dir: &str,
     impersonate_user: &Option<String>,
+    is_local: bool,
 ) -> Result<Script> {
-    let tokens = emit_commandline(app_instance, manifests_dir, impersonate_user);
+    let tokens = emit_commandline(app_instance, manifests_dir, impersonate_user, is_local);
     Ok(Script::from_vec(tokens))
 }
 
 pub fn emit_commandline(
     app_instance: &AppInstance,
     manifests_dir: &str,
     impersonate_user: &Option<String>,
+    is_local: bool,
 ) -> Vec<String> {
-    let mut cli = vec![
-        "kubectl",
-        "apply",
-        "-f",
-        manifests_dir,
-        "-n",
-        &app_instance.namespace_any(),
-        "--server-side",
-        "--prune",
-        "--applyset",
-        &app_instance.name_any(),
-        "--field-manager",
-        KUBIT_APPLIER_FIELD_MANAGER,
-        "--force-conflicts",
-        "-v=2",
-    ]
-    .iter()
-    .map(|s| s.to_string())
-    .collect::<Vec<_>>();
+    let mut cli: Vec<String> = vec![];
+
+    if is_local {
```

Comment:
I'm not a huge fan of this `if-else` branching here, but I think the "use containers" approach is a little more complicated than I first thought.

If we don't branch between local and running controller, we are then using "docker-in-docker". 

Perhaps we instead mount the Docker socket though and we're okay with that? Meaning we have sibling containers, as opposed to child containers (acts like d-in-d, but isn't d-in-d).

This is a part where I might've over-complicated it, I think if this is simplified then the overall approach may also be easier to implement too.

Reply:
most flags are shared. You can use a mutable vector, populate it with the right command depending on the is_local, and then adding the flags that are common between the two kinds of invocations

`let mut cli = Vec::new()`

and then `cli.extend(...)`

Reply:
I'll have to take another look at this, I might've messed it up initially.

This was my first intention, but using `.extend` was similar to an appending of arguments, so I thought the initial `cli`/`steps` vectors needed to be clean from the beginning.

I'll take another look at this approach though.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cd8ef11db1844f331f71c2e5a20d9dde

Reply:
This makes sense now upon revisiting it. Error was in my brain, not in the code :laughing: 

Splitting it out between common args, with different invocation methods (`docker` when local vs `kubecfg` in controller) makes a lot more sense.